grammar.js

```diff
@@ -106,12 +108,26 @@ module.exports = grammar({
 
     quote_escape: ($) => /`'/,
 
-    inline_quote: ($) =>
-      choice($._inline_quote, $._double_inline_quote, $._triple_inline_quote),
+    external_link: ($) => seq(
+        "[",
+        $.link,
+        "]"
+    ),
 
-    _inline_quote: ($) => seq("`", /[^']+/, "'"),
-    _double_inline_quote: ($) => seq("``", repeat(/([^`']|')/), "''"),
-    _triple_inline_quote: ($) => seq("```", repeat(/([^'\r\n]|')/), "'''"),
+    link: ($) => /(https?|ftp|file):\/\/[^\s\]]+/,
+
+    inline_quote: ($) =>
+      choice($._inline_quote, $._double_inline_quote),
+
+    _inline_quote: ($) => seq(
+        alias("`", $.quote_marker),
+        alias(/[^']+/, $.quote_content),
+        alias("'", $.quote_marker)),
+    _double_inline_quote: ($) => seq(
+        alias("``", $.quote_marker),
+        alias(repeat(/([^`']|')/), $.quote_content),
+        alias("''", $.quote_marker)),
+    // _triple_inline_quote: ($) => seq("```", repeat(/([^'\r\n]|')/), "'''"),
```

Comment:
Does this remove triple-tick code fences?

Reply:
I have a question about this. Seems you distinguish the triple quote into inline-quote and block-quote. However, this style of quote in edoc will be expanded to `<pre><![CDATA[...]]></pre>` in XHTML, meaning any content inside it is valid. An example from the erlang edoc manual:

```
     %% @doc ...as in the following code:
     %% ```f(X) ->
     %%       case X of
     %%          ...
     %%       end'''
```

And the <code>```lang</code> style is not defined by edoc. There is some conflict between them, I think.